drainer/config_test.go

```diff
@@ -118,3 +121,39 @@ func (t *testDrainerSuite) TestAdjustConfig(c *C) {
 	c.Assert(cfg.ListenAddr, Equals, "http://0.0.0.0:8257")
 	c.Assert(cfg.AdvertiseAddr, Equals, "http://192.168.15.12:8257")
 }
+
+func (t *testDrainerSuite) TestConfigParsingFileWithInvalidOptions(c *C) {
+	yc := struct {
+		DataDir                string `toml:"data-dir" json:"data-dir"`
+		ListenAddr             string `toml:"addr" json:"addr"`
+		AdvertiseAddr          string `toml:"advertise-addr" json:"advertise-addr"`
+		UnrecognizedOptionTest bool   `toml:"unrecognized-option-test" json:"unrecognized-option-test"`
+	}{
+		"data.drainer",
+		"192.168.15.10:8257",
+		"192.168.15.10:8257",
+		true,
+	}
+
+	var buf bytes.Buffer
+	e := toml.NewEncoder(&buf)
+	err := e.Encode(yc)
+	c.Assert(err, IsNil)
+
+	tmpfile, err := util.CreateCfgFile(buf.Bytes(), "drainer_config")
+	if tmpfile != nil && len(tmpfile.Name()) > 0 {
```

Comment:
I'm not so sure about this part. But I think it's necessary to delete tmpfile once this file was created, even if some error occured.

Reply:
You can use `c.Mkdir()` to create a temporary directory and let gocheck do it for you.

Reply:
Revised yet.